lib/site_prism/dsl.rb

```diff
@@ -228,27 +222,31 @@ def create_error_method(name)
       end
 
       def deduce_iframe_scope_find_args(args)
+        warn_on_invalid_selector_input(args)
         case args[0]
-        when Integer
-          [args[0]]
-        when String
-          [:css, args[0]]
-        else
-          args
+        when Integer then [args[0]]
+        when String  then [:css, args[0]]
+        else args
         end
       end
 
       def deduce_iframe_element_find_args(args)
+        warn_on_invalid_selector_input(args)
         case args[0]
-        when Integer
-          "iframe:nth-of-type(#{args[0] + 1})"
-        when String
-          [:css, args[0]]
-        else
-          args
+        when Integer then "iframe:nth-of-type(#{args[0] + 1})"
+        when String  then [:css, args[0]]
+        else args
         end
       end
 
+      def warn_on_invalid_selector_input(args)
```

Comment:
Actually, why can't we use xpath? As long as you have something like:

```ruby
page.driver.browser.switch_to.frame
```
You should be able to use the selector of choice.

Reply:
I see, yes, they should be defining ```:xpath``` as the selector if they choose to need it.

Reply:
I would merge it for you, but I don't have access to.